### DISCUSSION TITLE
Rating for duplicate mahjong

### DISCUSSION THREAD

**Discussion by Cryolite** [smly/mjai.app#14] [General]
As of 2022.10.11 (Tue) Release (#13), the ranks on which the rating changes are based seem to be determined by the average score of the games of duplicate mahjong. However, I think that this way of calculating the rank may not reflect well the typical objective of normal mahjong and thus mahjong AIs.

The typical objective of normal mahjong and mahjong AIs is to achieve the highest possible rank in a single game. In ranking matches in Tenhou (天鳳), only the final rank of a single game is the basis for calculating the rating. In ranking matches in Mahjong Soul (雀魂), most part of rating changes is determined by the final rank of a single game. Accordingly, most mahjong AIs are also trained to maximize the final rank (or almost equivalent objective) in a single game.

In maximizing the rank in a single game, it is possible to use an advanced tactic that dares to sacrifice scores. This is so-called "差し込み". Therefore, I think that using the rank determined by the average of scores from multiple games in duplicate mahjong as the basis for rating change is a little out of line with the original objective of mahjong.

Here, my specific proposal is to use the rank determined by the average of the final ranks of multiple games in duplicate mahjong (average rank in duplicate mahjong) as the basis for rating change.

I would appreciate your consideration of the above change.

**Comment by smly** [comment]
Thanks for the feedback for improvement! It was not clearly explained in the release notes. The average score of the games of duplicate mahjong is **not** used for the rating changes. The average score in the list of matches is misleading. It should be improved.

Rating changes are made every game. Only the final rank for each game is used for calculating the rating. Duplicate mahjong is played in 4 games, so it is updated 4 times. I think this method is good for making the difference between losing to a weaker player and losing to a stronger player in multiple matches. How do you think about this?

**Reply by smly** [reply]
All feedback is very helpful to know what users care about and what needs to be clarified in the documentation. Thanks again!

**Reply by Cryolite** [reply]
Oh, I misunderstood. Your explanation of how the rating changes almost makes sense to me.

My only concern is that the rating changes at each game in duplicate mahjong. This implies that the rating difference between players, which is the basis for the calculation of rating changes, changes for each game in duplicate mahjong, doesn't it? If so, considering the intent of duplicate mahjong, should the rating change in each game in duplicate mahjong be calculated using the rating difference between the players BEFORE they start playing duplicate mahjong? However, the difference in rating changes is not expected to be very large.

**Reply by smly** [reply]
I totally agree with you on the calculation of rating changes. The current system calculates the rating change based on the rating before they start playing duplicate mahjong as you mentioned. Also, it would be nice to see the rating change for each of 4 games in the log viewer. I will implement it soon.
